### ISSUE TITLE
Setting a basic Bearer token

### ISSUE THREAD

**Issue by mtownsend5512** [Ricorocks-Digital-Agency/Soap#38]
First, let me say that I've been working on this on and off for over a week. I'm baffled by why I can't get it working. My token works correctly when used through Postman, but I can't get the package to utilize the header/token correctly.

I am instantiating the class and trying to set it as a global header like so:

`Soap::to($url)->trace()->headers(Soap::header()->name('Authorization')->data('Bearer ' . $token))->withOptions($options)->call($method, $payload);`

I get the following error:

`SOAP-ERROR: Parsing WSDL: Couldn't load from '<MY_SOAP_URL> : failed to load external entity "<MY_SOAP_URL>"`

What am I doing wrong here? I'm just trying to set a simple `Authorization` `Bearer <token>` header.

**Comment by lukeraymonddowning** [comment]
Can you try the same request with the standard PHP soap client and let me know if it works?

**Comment by veewee** [comment]
Hello @mtownsend5512 

There is a difference between soap headers and HTTP headers.
In this case you are trying to set SOAP headers.

You could try to add the header by using  a `stream_context` through the `SoapClient`'s `$options`.
See https://www.php.net/manual/en/context.http.php
But I think that context is not being used by the SoapClient for fetching the WSDL.

Leaving you with the option to either download the file(s) locally or fetch it dynamically (with curl or a stream wrapper).

**Comment by mtownsend5512** [comment]
Yeah, so it does work when you kick it through regular curl, but I've tried a few different variations of the `stream_context` with no luck. I wonder, is this something relatively painless to support?

Although SOAP is antiquated in modern apps, large enterprise apps like Microsoft are still using it for Business Central and they are depreciating user/pass authentication in favor of Bearer tokens.

**Comment by veewee** [comment]
It is a known bug in the existing soap-client.
Several bugs have been reported, like https://bugs.php.net/bug.php?id=47761

As mentioned in previous comment, You can get around the issue by downloading the wsdl first.

You might want to use one of these to do the work for you:

https://github.com/php-soap/ext-soap-engine#wsdlprovider
https://github.com/php-soap/wsdl#wsdl-loader

(These could also be included in here some day to make it less painful to deal with)

**Comment by lukeraymonddowning** [comment]
Closing this right now until we decide on #35
